### PR TITLE
Fix Term

### PR DESCRIPTION
**Pull request by Parsifal22**
Updated the term. Now it works as written on paper

### CODE REVIEW

File: cyraft/node.py [hunk 1]
```diff
@@ -685,12 +717,22 @@ async def _on_term_timeout(self) -> None:
             self._term,
         )
         assert self._state == RaftState.LEADER, "Only leaders have a term timeout"
-        self._term += 1
```
Comment (maksimdrachov) [inline cyraft/node.py:688]: As far as I can tell, this is the only relevant change, all other changes are related to leader election.

(Not talking about small fixes/log changes)

File: cyraft/node.py [hunk 2]
```diff
@@ -203,43 +203,48 @@ async def _serve_request_vote_impl(
...
-            c["request_vote"] + "Node ID: %d -- Request vote request denied (unknown reason)" + c["end_color"],
-            self._node.id,
-        )
-        _logger.info(
-            c["request_vote"] + "Node ID: %d -- Current term: %d, Request term: %d" + c["end_color"],
-            self._node.id,
-            self._term,
-            request.term,
-        )
-        assert False, "Should not reach here!"
```
Comment (maksimdrachov) [inline cyraft/node.py:242]: `_serve_request_vote_impl` is not related to this term issue?

File: cyraft/node.py [hunk 3]
```diff
@@ -380,25 +397,8 @@ def _append_entries_processing(
...
-                _logger.info(
-                    c["append_entries"] + "Node ID: %d -- deleting from: %d" + c["end_color"],
-                    self._node.id,
-                    log_index + 1,
-                )
-                number_of_entries_to_delete = len(self._log) - (log_index + 1)
-                self._next_index = [x - number_of_entries_to_delete for x in self._next_index]
-                del self._log[log_index + 1 :]
-                self._commit_index = log_index
-                break
```
Comment (maksimdrachov) [inline cyraft/node.py:401]: `_append_entries_processing` does not require this change, as discussed [here](https://github.com/OpenCyphal-Garage/cyraft/pull/11#discussion_r1682389382).

File: tests/raft_node.py [hunk 4]
```diff
@@ -95,40 +95,9 @@ async def _unittest_raft_node_init() -> None:
...
-    await asyncio.sleep(TERM_TIMEOUT)  # + 0.1 to make sure the timer has been reset
-    assert raft_node._term == 1
-    await asyncio.sleep(TERM_TIMEOUT)
-    assert raft_node._term == 2
-    await asyncio.sleep(TERM_TIMEOUT)
-    assert raft_node._term == 3
-
-    raft_node._change_state(RaftState.FOLLOWER)  # follower should not increase term
-    await asyncio.sleep(TERM_TIMEOUT)
-    assert raft_node._term == 3
```
Comment (maksimdrachov) [inline tests/raft_node.py:128]: Removing is not necessary, adjust it so it tests that once the Leader node is elected the term is not increased, regardless of how many term timeouts would happen.

File: tests/raft_node.py [hunk 5]
```diff
@@ -519,152 +472,18 @@ async def _unittest_raft_node_append_entries_rpc() -> None:
...
-    assert raft_node._log[1].entry.name.value.tobytes().decode("utf-8") == "top_1"
-    assert raft_node._log[1].entry.value == 7
-    assert raft_node._log[2].term == 8
-    assert raft_node._log[2].entry.name.value.tobytes().decode("utf-8") == "top_2"
-    assert raft_node._log[2].entry.value == 11
-    assert raft_node._log[3].term == 9
-    assert raft_node._log[3].entry.name.value.tobytes().decode("utf-8") == "top_3"
-    assert raft_node._log[3].entry.value == 12
-
-    _logger.info("================== TEST 4: Add an already existing log entry ==================")
```
Comment (maksimdrachov) [inline tests/raft_node.py:616]: This is related to `append_entries_processing`, please restore it.

File: tests/raft_node.py [hunk 6]
```diff
@@ -519,152 +472,18 @@ async def _unittest_raft_node_append_entries_rpc() -> None:
...
-
-    new_entry = sirius_cyber_corp.LogEntry_1(
-        term=10,
-        entry=sirius_cyber_corp.Entry_1(
-            name=uavcan.primitive.String_1(value="top_4"),
-            value=13,
-        ),
-    )
-    request = sirius_cyber_corp.AppendEntries_1.Request(
-        term=10,
```
Comment (maksimdrachov) [inline tests/raft_node.py:667]: Why did you remove this part?

**Review by maksimdrachov — CHANGES_REQUESTED** [review]

File: tests/raft_node.py (hunk 5 above)
Comment (Parsifal22) [reply tests/raft_node.py:616]: These tests simply won't pass because in this [part of the code](https://github.com/OpenCyphal-Garage/cyraft/pull/16/files/b58d3c8e7e8a2be287df2a8e355704ad086fda78#diff-afe050a14d92bccbfdd641958448655eb3e4d834d0dce08f5dc99f72b938600b:~:text=for%20log_index%2C,break). It changes values ​​unless the value of `log_entry.term` matches the value of `request.log_entry[0].term`. `request.log_entry[0].term` will be different only if the election happens, but that doesn't happen in these tests. The leader can no longer change the logs, as they have a static term and these tests do not fit the new conditions.

Comment (maksimdrachov) [reply tests/raft_node.py:616]: [It is possible.](https://github.com/OpenCyphal-Garage/cyraft/pull/17)

image
